### PR TITLE
SyncedAt Column for QRep

### PR DESCRIPTION
**Pull request by Amogh-Bharadwaj**
Implements `_PEERDB_SYNCED_AT` for Query Replication mirrors for PG -> [PG, BQ, SF]

### CODE REVIEW

File: flow/connectors/postgres/qrep_sync_method.go [hunk 1]
```diff
@@ -81,14 +82,27 @@ func (s *QRepStagingTableSync) SyncQRepRecords(
...
+		}
 	} else {
 		// Step 2.1: Create a temp staging table
 		stagingTableName := fmt.Sprintf("_peerdb_staging_%s", shared.RandomString(8))
 		stagingTableIdentifier := pgx.Identifier{s.connector.metadataSchema, stagingTableName}
 		dstTableIdentifier := pgx.Identifier{dstTableName.Schema, dstTableName.Table}
 
 		createStagingTableStmt := fmt.Sprintf(
-			"CREATE UNLOGGED TABLE %s (LIKE %s);",
+			"CREATE UNLOGGED TABLE %s (LIKE %s INCLUDING INDEXES);",
```
Comment (serprex) [inline flow/connectors/postgres/qrep_sync_method.go:105]: Is this related or just an optimization realized while implementing?

File: flow/connectors/postgres/qrep_sync_method.go [hunk 2]
```diff
@@ -128,16 +142,19 @@ func (s *QRepStagingTableSync) SyncQRepRecords(
...
+			fmt.Sprintf(`"%s" = CURRENT_TIMESTAMP`, syncedAtCol))
 		setClause := strings.Join(setClauseArray, ",")
-		selectStr := strings.Join(selectStrArray, ",")
+		selectStrColsSQL := strings.Join(append(selectStrArray,
+			fmt.Sprintf(`"%s"`, syncedAtCol)), ",")
+		selectStrValuesSQL := strings.Join(append(selectStrArray, `CURRENT_TIMESTAMP`), ",")
 
 		// Step 2.3: Perform the upsert operation, ON CONFLICT UPDATE
 		upsertStmt := fmt.Sprintf(
 			"INSERT INTO %s (%s) SELECT %s FROM %s ON CONFLICT (%s) DO UPDATE SET %s;",
```
Comment (serprex) [inline flow/connectors/postgres/qrep_sync_method.go:154]:
```suggestion
			"INSERT INTO %s (%s, "%s") SELECT %s, CURRENT_TIMESTAMP FROM %s ON CONFLICT (%s) DO UPDATE SET %s;",
```
& then pass `syncedAtCol` as an extra format param. Avoids aliased appends

File: flow/e2e/postgres/qrep_flow_pg_test.go [hunk 3]
```diff
@@ -134,6 +135,30 @@ func (s *PeerFlowE2ETestSuitePG) compareQuery(srcSchemaQualified, dstSchemaQuali
...
+		if err != nil {
+			return err
+		}
+
+		if !syncedAt.Valid {
+			return fmt.Errorf("synced_at is not valid")
+		}
+	}
+
+	if rows.Err() != nil {
```
Comment (serprex) [inline flow/e2e/postgres/qrep_flow_pg_test.go:156]:
```suggestion
	return rows.Err()
```

File: flow/connectors/postgres/qrep_sync_method.go (hunk 1 above)
Comment (Amogh-Bharadwaj) [reply flow/connectors/postgres/qrep_sync_method.go]: It isn't needed. Initially thought it was

**Review by iskakaushik — APPROVED** [review]

**Pull request merged**